test/tests/functional/pbs_alps_request_tunables.py

```diff
@@ -0,0 +1,150 @@
+        after the set interval.
+        """
+        ari = 4.298
+        self.mom.add_config({'$alps_release_interval': ari})
+
+        # submit a job and then delete it after it starts running
+        start_time = time.time()
+        j1 = Job(TEST_USER)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1)
+
+        # Look for a message that confirms that reservation is deleted
+        self.mom.log_match("%s;ALPS reservation cancelled" % jid1,
+                           starttime=start_time, max_attempts=5,
+                           interval=2)
+        # Now that we know that reservation is cleared we should
+        # check for time difference between each cancellation request
+        out = self.mom.log_match("%s;Canceling ALPS reservation *" % jid1,
+                                 n='ALL', max_attempts=5, regexp=True,
+                                 allmatch=True)
+
+        # We found something, Let's first check there are atleast 2 such
+        # log messages, If not then that means reservation was cancelled
+        # in the first attempt itself, at that point right thing to do is
+        # to either run it again or find out a way to delay the reservation
+        # cancellation at ALPS level itself.
+        if len(out) >= 2:
+            nline_prev, line_prev = out[0]
+            time_prev = self.get_epoch(line_prev)
+            for data in out[1:]:
+                nline, line_current = data
+                time_current = self.get_epoch(line_current)
+                self.assertGreaterEqual(time_current - time_prev,
+                                        math.ceil(ari),
+                                        msg="alps_release_interval not working"
+                                        )
+                time_prev = time_current
+
+        else:
+            self.skipTest("Reservation cancelled without retry, Try again!")
+
+    def test_alps_release_jitter(self):
+        """
+        Set alps_release_jitter to a higher value and then notice that
+        subsequest reservation cancellation requests are made by adding
+        a random time interval (less than jitter) to alps_release_interval.
+        """
+        arj = 2.198
+        ari = 1
+        max_delay = (ari + math.ceil(2.198))
+        self.mom.add_config({'$alps_release_jitter': arj})
+        self.mom.add_config({'$alps_release_interval': ari})
+        # There is no good way to test jitter and it is a random number
+        # less than value set in alps_release_jitter. So in this case
+        # we can probably try deleting a reservation a few times.
+        for each in range(5):
+            # submit a job and then delete it after it starts running
+            start_time = time.time()
+            j1 = Job(TEST_USER)
+            jid1 = self.server.submit(j1)
+            self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
```

Comment:
To ensure that there is enough time for the aprun to get started, maybe we should sleep for a couple seconds before we delete the job? The sleep will likely also cause the ALPS reservation not to be freed right away so that you know you are getting into the code that uses jitter.